metrics-operator/converter/common_test.go

```diff
@@ -19,3 +19,68 @@ func TestIsGreaterOrEqual(t *testing.T) {
 	require.True(t, isGreaterOrEqual(">"))
 	require.True(t, isGreaterOrEqual(">="))
 }
+
+func TestConvertResourceName(t *testing.T) {
+	type args struct {
+		name string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
```

Comment:
Can we introduce also the following cases:

- 253 characters long name
- empty string
- string with leading/trailing '-'
- string with more than 253 chars